src/Analyzers/Core/Analyzers/UseConditionalExpression/ForAssignment/UseConditionalExpressionForAssignmentHelpers.cs

```diff
@@ -132,6 +153,40 @@ static bool ContainsLocalReference(HashSet<ILocalSymbol> declaredPatternSymbols,
 
             return false;
         }
+
+        static IEnumerable<IOperation> GetNullCheckedExpressions(IOperation operation)
+        {
+            foreach (var current in operation.DescendantsAndSelf())
+            {
+                // x != null  is a null check of x
+                if (current is IBinaryOperation { OperatorKind: BinaryOperatorKind.Equals or BinaryOperatorKind.NotEquals } binaryOperation)
+                {
+                    if (binaryOperation.LeftOperand.ConstantValue is { HasValue: true, Value: null })
+                    {
+                        yield return binaryOperation.RightOperand;
+                    }
+                    else if (binaryOperation.RightOperand.ConstantValue is { HasValue: true, Value: null })
+                    {
+                        yield return binaryOperation.LeftOperand;
+                    }
+                }
+                else if (current is IIsPatternOperation isPatternOperation)
+                {
+                    // x is Y y    is a null check of x
+                    yield return isPatternOperation.Value;
+                }
+                else if (current is IIsTypeOperation isTypeOperation)
+                {
+                    // x is Y    is a null check of x
+                    yield return isTypeOperation.ValueOperand;
+                }
+            }
+
+            yield break;
```

Comment:
>yield break; [](http://example.com/codeflow?start=12&length=12)

Curious what the yield break provides here

Reply:
nothing whatsoever :D   i started with that tehre originally as i TDDed the impl of this.  and i never removed.  can do so in followup :)